### PR TITLE
Add method to get ios 9.3 web inspector socket

### PR DESCRIPTION
**Pull request by imurchie**
In the first step of providing support for iOS 11.3 web, we need to find the unix domain socket for the simulator.

I've searched _ever_ file on my machine while a sim is running, and can find no reference to the correct socket. The best solution I've come up with is ot use `lsof` and hope that the grouping together of entires for each sim (this works in my experimentation).

### CODE REVIEW

File: lib/simulator-xcode-6.js [hunk 1]
```diff
@@ -1131,6 +1131,11 @@ class SimulatorXcode6 extends EventEmitter {
     log.debug(`Final device string is '${iosDeviceString}'`);
     return iosDeviceString;
   }
+
+  async getWebInspectorSocket () {
```
Comment (mykola-mokhnach) [inline lib/simulator-xcode-6.js]: it would be nice to have some documentation for this method

File: lib/simulator-xcode-9.3.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    //   launchd_s 69760 isaac    8u  unix 0x57aa4fceea39598b      0t0      /private/tmp/com.apple.launchd.2j5k1TMh6i/com.apple.webinspectord_sim.socket
+    //   launchd_s 69760 isaac    9u  unix 0x57aa4fceea394c43      0t0      /private/tmp/com.apple.launchd.4zm9JO9KEs/com.apple.testmanagerd.unix-domain.socket
+    //   launchd_s 69760 isaac   10u  unix 0x57aa4fceea395f03      0t0      /private/tmp/com.apple.launchd.2B2u8CkN8S/Listeners
+    //   launchd_s 69760 isaac   11u  unix 0x57aa4fceea39598b      0t0      /private/tmp/com.apple.launchd.2j5k1TMh6i/com.apple.webinspectord_sim.socket
+    //   launchd_s 69760 isaac   12u  unix 0x57aa4fceea394c43      0t0      /private/tmp/com.apple.launchd.4zm9JO9KEs/com.apple.testmanagerd.unix-domain.socket
+    // these _appear_ to always be grouped together (so, the records for the particular sim are all in a group, before the next sim, etc.)
+    // so starting from the correct UDID, we ought to be able to pull the next record with `com.apple.webinspectord_sim.socket` to get the correct socket
+    let {stdout} = await exec('lsof', ['-aUc', 'launchd_sim']);
+    for (let record of stdout.split('com.apple.CoreSimulator.SimDevice.')) {
```
Comment (mykola-mokhnach) [inline lib/simulator-xcode-9.3.js]: does it work if there are zero sockets in the output?

File: lib/simulator-xcode-9.3.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        continue;
+      }
+      // console.log('we got the record for the udid');
+      // console.log(record);
+      // https://regex101.com/r/MEL55t/1
+      const match = /\s+(\S+com\.apple\.webinspectord_sim\.socket)/.exec(record);
+      if (!match) {
+        return null;
+      }
+      this.webInspectorSocket = match[1];
```
Comment (mykola-mokhnach) [inline lib/simulator-xcode-9.3.js]: shall we also verify the matched file really exists?

File: lib/simulator.js [hunk 4]
```diff
@@ -50,7 +51,11 @@ async function getSimulator (udid) {
   } else if (xcodeVersion.major === 8) {
     return new SimulatorXcode8(udid, xcodeVersion);
   } else if (xcodeVersion.major === 9) {
-    return new SimulatorXcode9(udid, xcodeVersion);
+    if (xcodeVersion.minor < 3) {
+      return new SimulatorXcode9(udid, xcodeVersion);
+    } else {
```
Comment (mykola-mokhnach) [inline lib/simulator.js]: else is not needed

File: lib/simulator-xcode-9.3.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      return this.webInspectorSocket;
+    }
+  }
+}
```
Comment (mykola-mokhnach) [inline lib/simulator-xcode-9.3.js]: missing `return null` ?

File: lib/simulator-xcode-9.3.js (hunk 2 above)
Comment (imurchie) [reply lib/simulator-xcode-9.3.js]: Yes. Returns `null`.

File: lib/simulator-xcode-6.js [hunk 6]
```diff
@@ -1132,6 +1132,10 @@ class SimulatorXcode6 extends EventEmitter {
     return iosDeviceString;
   }
 
+  /*
+   * @return {string} The full path to the simulator's WebInspector Unix Domain Socket
```
Comment (mykola-mokhnach) [inline lib/simulator-xcode-6.js]: `{?string}` would be more precise

File: lib/simulator-xcode-6.js [hunk 7]
```diff
@@ -1131,6 +1131,15 @@ class SimulatorXcode6 extends EventEmitter {
     log.debug(`Final device string is '${iosDeviceString}'`);
     return iosDeviceString;
   }
+
+  /*
+   * @return {string} The full path to the simulator's WebInspector Unix Domain Socket
+  *                   or `null` if there is no socket.
```
Comment (mykola-mokhnach) [inline lib/simulator-xcode-6.js]: format

**Pull request merged**